metadata-ws/src/test/java/uk/ac/ebi/ampt2d/metadata/cucumber/CommonSteps.java

```diff
@@ -0,0 +1,349 @@
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@Ignore
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class CommonSteps {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private AnalysisRepository analysisRepository;
+
+    @Autowired
+    private ReferenceSequenceRepository referenceSequenceRepository;
+
+    @Autowired
+    private FileRepository fileRepository;
+
+    @Autowired
+    private SampleRepository sampleRepository;
+
+    @Autowired
+    private StudyRepository studyRepository;
+
+    @Autowired
+    private TaxonomyRepository taxonomyRepository;
+
+    @Autowired
+    private WebResourceRepository webResourceRepository;
+
+    @Before
+    public void cleanDatabases() {
+        analysisRepository.deleteAll();
+        referenceSequenceRepository.deleteAll();
+        fileRepository.deleteAll();
+        sampleRepository.deleteAll();
+        studyRepository.deleteAll();
+        taxonomyRepository.deleteAll();
+        webResourceRepository.deleteAll();
+    }
+
+    @Before
+    public void cleanStates() {
+        CommonStates.clear();
+    }
+
+    /* perform http request */
+
+    @When("^user request GET ([\\S]*)$")
+    public void performGetOnResourceUri(String resourceUri) throws Exception {
+        CommonStates.setResultActions(mockMvc.perform(get(resourceUri)));
+    }
+
+    @When("^user request GET with value of (.*)$")
+    public void performGetWithResourceUriKey(String resourceUriKey) throws Exception {
+        CommonStates.setResultActions(mockMvc.perform(get(CommonStates.getUrl(resourceUriKey))));
+    }
+
+    @When("^user request GET for (.*) of (.*)$")
+    public void performGetForLinkedObjects(String className, String resourceUriKey) throws Exception {
+        CommonStates.setResultActions(mockMvc.perform(get(CommonStates.getUrl(resourceUriKey)+"/"+className)));
+    }
+
+    @When("^user request POST (.*) with json data:$")
+    public void performPostOnResourceUriWithJsonData(String resourceUri, String jsonData) throws Exception {
+        CommonStates.setResultActions(mockMvc.perform(post(resourceUri)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(jsonData.getBytes())));
+    }
+
+    @When("^user request PATCH (.*) with list (.*) for (.*)")
+    public void performPatchOnResourceWithLinkedObject(String urlKey, String linkedObjectUrlKeys,
+                                                       String linkedObjectClassName) throws Exception {
+        List<String> newUrls = null;
+        if (!linkedObjectUrlKeys.equals("NONE")) {
```

Comment:
OK, after futher discussion with @iyangar this can not call `CommonStates.getUrls` as it is directly. The difference between them are one is filtering on the null URL with non-existing key while the other is not. The use case for this method is to for the scenario like `update an analysis with invalid reference sequences should fail`. One of considerations there is to update an analysis with a list reference sequences that contains a `null` object.
Since this is the only place where this kind of behaviour is expected, I am happy to leave it as it is.